### PR TITLE
Basic UDP socket

### PR DESCRIPTION
**Pull request by xymus**
This PR intro `UDPSocket` and cleans up a bit of the native layer of socket to avoid using callbacks when possible.

`UDPSocket` support sending messages to remote sockets without a connection and more importantly receive message from any sender. However, as of yet it does not support connections or the stream API of Nit.

Some TODO with the sockets:
* Stop using `gethostbyname` :neutral_face: 
* Decide on one or many APIs. The native layer has an API closed to the C API and to what is available in other languages, maybe it can be clean up further and exposed as is.

---
Please ignore the first 2 commits from #1573.

### CODE REVIEW

**Comment by privat** [comment]
+1

**Comment by BlackMinou** [comment]
+1

File: lib/socket/socket.nit [hunk 1]
```diff
@@ -20,11 +20,28 @@ module socket
...
+	private var native: NativeSocket is noinit
+
+	# Is this socket closed?
+	var closed = false
+
+	# Set whether calls to `accept` are blocking
+	fun blocking=(value: Bool)
+	do
+		# We use the opposite from the native version as the native API
+		# is closer to the C API. In the Nity API, we use a positive version
```
Comment (Morriar) [inline lib/socket/socket.nit]: Nity API?

**Comment by Morriar** [comment]
+1

**Pull request merged**